### PR TITLE
Only add open path when module exposes types or bindings.

### LINKED ISSUE #13761: Signature file generation does not handle types from a shared namespace
When generating a signature file from a type annotation with a type from a shared namespace, the generated code requires an additional inclusion of the shared namespace.

**Repro steps**

```fsharp
namespace Foo.Types

type Area = | Area of string * int

namespace Foo.Other

type Map<'t,'v> =
    member this.Calculate : Foo.Types.Area = failwith "todo"
```

leads to

```fsharp
namespace Foo.Types
    
    type Area = | Area of string * int
namespace Foo.Other
    
    [<Class>]
    type Map<'t,'v> =
        
        member Calculate: Types.Area
```

**Expected behaviour**

The generated signature should be considered equivalent to the backing source file.

**Actual behaviour**

The generated code isn't valid:

image

**Known workarounds**

Edit signature file by hand.

### PR DESCRIPTION
**Pull request by nojaf**
I believe this fixes #13761.

### CODE REVIEW

File: src/Compiler/Checking/NicePrint.fs [hunk 1]
```diff
@@ -2326,12 +2326,20 @@ module InferredSigPrinting =
...
             let denv =
-                innerPath
-                |> List.choose (fun (path, kind) ->
-                    match kind with
-                    | ModuleOrNamespaceKind.Namespace false -> None
-                    | _ -> Some path)
-                |> denv.AddOpenPath
+                match def with
+                | ModuleOrNamespaceContents.TMDefRec (tycons = []; bindings = bindings) ->
+                    let inline isModule b =
```
Comment (dsyme) [inline src/Compiler/Checking/NicePrint.fs:2331]: Why inline?

**Comment by dsyme** [comment]
The fix seems arrange. It feels like a reimplementation of isConcreteNamespace. But I need to look more carefully,  just looking on my phone right now.

File: src/Compiler/Checking/NicePrint.fs (hunk 1 above)
Comment (nojaf) [reply src/Compiler/Checking/NicePrint.fs]: Yeah, should not have been there.

**Comment by nojaf** [comment]
Yes, I believe I can re-use `isConcreteNamespace` to get the same result.

**Pull request merged**